### PR TITLE
FPGA: new reference design and linear algebra library - Cholesky-based matrix inversion

### PR DESCRIPTION
**Pull request by yuguen-intel**
# Adding a New Sample

## Description

This pull request adds both:
- a new linear algebra header library (streaming_cholesky_inversion.hpp) that performs matrix inversions based on the Cholesky decomposition.
- a new reference design that exercises this new library.

## Checklist
Administrative
- [x] Review sample design with the appropriate [Domain Expert](https://github.com/oneapi-src/oneAPI-samples/wiki/Reviewers-and-Domain-Experts): @mdbtucker

Code Development
- [x] Implement coding guidelines and ensure code quality. [see wiki for details](https://github.com/oneapi-src/oneAPI-samples/wiki/General-Code-Guidelines)
- [x] Adhere to readme template 
- [x] Enforce format via clang-format config file
- [x] Adhere to sample.json specification. https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-specification
- [x] Ensure/create CI test configurations for sample (ciTests field) https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-ci-test-object
- [x] Run jsonlint on sample.json to verify json syntax. www.jsonlint.com

Security and Legal
- [ ] OSPDT Approval (see Project Manager for assistance)
- [x] Compile using the following compiler flags and fix any warnings, the falgs are: "/Wall -Wformat-security -Werror=format-security"
- [ ] Virus scan

Review
- [ ] Review DPC++ code with Paul Peterseon. (GitHub User: pmpeter1)
- [ ] Review readme with Tom Lenth(@tomlenth) and/or Project Manager
- [ ] Tested using Dev Cloud when applicable

### CODE REVIEW

File: DirectProgramming/DPC++FPGA/ReferenceDesigns/cholesky_inversion/src/cholesky_inversion_demo.cpp [hunk 1]
```diff
@@ -0,0 +1,425 @@
...
+float RandomValueInInterval(float min, float max) {
+  return min + static_cast<float>(rand()) /
+                   (static_cast<float>(RAND_MAX) / (max - min));
+}
+
+/*
+  Generate the input matrices for the cholesky inversion
+*/
+template <int matrices_to_invert, int matrix_size, int columns, typename T>
+void generateInputData(std::vector<T> &a_matrix) {
```
Comment (pmpeter1) [inline DirectProgramming/DPC++FPGA/ReferenceDesigns/cholesky_inversion/src/cholesky_inversion_demo.cpp:66]: Inconsistent naming convention for Function.

File: DirectProgramming/DPC++FPGA/ReferenceDesigns/cholesky_inversion/src/cholesky_inversion_demo.cpp [hunk 2]
```diff
@@ -0,0 +1,425 @@
...
+
+  }  // end of mat_idx
+}
+
+/*
+  Check results for correctness
+*/
+template <int inverted_matrices, int matrix_size, int invert_matrix_size,
+          int rows, int columns, typename T>
+int checkResults(std::vector<T> &a_matrix, std::vector<T> &i_matrix) {
```
Comment (pmpeter1) [inline DirectProgramming/DPC++FPGA/ReferenceDesigns/cholesky_inversion/src/cholesky_inversion_demo.cpp:183]: Inconsistent naming convention for Function.

**Review by pmpeter1 — CHANGES_REQUESTED** [review]

**Review by tomlenth — CHANGES_REQUESTED** [review]
the second paragraph with the reference to documentation has a couple of <br> tags.  these don't always render, but should be removed.

**Comment by yuguen-intel** [comment]
@pmpeter1 thanks, fixed now
@tomlenth do you mean the <br> tags? These appear in all FPGA READMEs. If you think these should be removed, we can make a bulk change to all the FPGA READMEs rather than having it fixed it but not in the other ones?

**Comment by yuguen-intel** [comment]
@tomlenth I'll do a bulk change then!
@praveenkk123 Can you have a look at this PR? It has been reviewed internally before, and I've manually validated it

**Review by praveenkk123 — APPROVED** [review]

**Review by pmpeter1 — APPROVED** [review]

**Pull request merged**